internal/common/resource.go

```diff
@@ -19,10 +19,15 @@ type ResourceStatus struct {
 	Degraded     StatusMessage
 }
 
-type ReconcileFunc = func(*Request) (ResourceStatus, error)
+type ReconcileResult struct {
+	ResourceStatus
+	OperationResult controllerutil.OperationResult
+}
```

Comment:
Can you also move `Resource` field from `ResourceStatus` struct to this struct?
I think it would make more sense. Then the status `ResourceStatus` would contain only status messages and all the other information will be in `ReconcileResult`.

And can you give name to the inlined `ResourceStatus` struct? For example: `status`.
Inlining a struct would be useful if we wanted to call methods defined on the inlined struct. But the `ResourceStatus` does not have any methods.

Reply:
Right this is far better.